tests/unit/views/notifications/test_project_crew_notification.py

```diff
@@ -0,0 +1,200 @@
+            actor=twoflower_member.granted_by.fullname,
+            project=twoflower_member.project.joined_title,
+            user=twoflower_member.user.fullname,
+        )
+        == notification_string
+    )
+
+
+@when(
+    parsers.parse("Vetinari invites Twoflower to the project with a role {role}"),
+    target_fixture='twoflower_member',
+)
+def invite_twoflower_member(
+    role,
+    db_session,
+    client,
+    login,
+    user_rincewind,
+    user_twoflower,
+    project_expo2010,
+    user_vetinari,
+) -> models.ProjectCrewMembership:
+    roles = [_r.strip() for _r in role.split(',')]
+    is_editor = 'editor' in roles
+    is_promoter = 'promoter' in roles
+    is_usher = 'crew' in roles
+    twoflower_member = models.ProjectCrewMembership(
+        parent=project_expo2010,
+        user=user_twoflower,
+        is_editor=is_editor,
+        is_promoter=is_promoter,
+        is_usher=is_usher,
+        granted_by=user_vetinari,
+    )
+    twoflower_member.record_type = MEMBERSHIP_RECORD_TYPE.INVITE
+    db_session.add(twoflower_member)
+    db_session.commit()
+    return twoflower_member
+
+
+@when(
+    parsers.parse(
+        "Twoflower accepts the invitation to be an editor of project Expo 2010 with a role {role}"
+    ),
+    target_fixture='twoflower_member',
+)
+def accept_twoflower_member(
+    role,
+    db_session,
+    client,
+    login,
+    user_rincewind,
+    user_twoflower,
+    project_expo2010,
+    user_vetinari,
+) -> models.ProjectCrewMembership:
+    roles = [_r.strip() for _r in role.split(',')]
+    is_editor = 'editor' in roles
+    is_promoter = 'promoter' in roles
+    is_usher = 'crew' in roles
+    twoflower_member = models.ProjectCrewMembership(
+        parent=project_expo2010,
+        user=user_twoflower,
+        is_editor=is_editor,
+        is_promoter=is_promoter,
+        is_usher=is_usher,
+        granted_by=user_vetinari,
+    )
+    twoflower_member.record_type = MEMBERSHIP_RECORD_TYPE.ACCEPT
+    db_session.add(twoflower_member)
+    db_session.commit()
+    return twoflower_member
+
+
+@when(
+    parsers.parse("Twoflower's role changes to {role}"),
+    target_fixture='twoflower_member',
+)
+def amend_twoflower_member(
+    role,
+    db_session,
+    client,
+    login,
+    user_rincewind,
+    user_twoflower,
+    project_expo2010,
+    user_vetinari,
+) -> models.ProjectCrewMembership:
+    roles = [_r.strip() for _r in role.split(',')]
+    is_editor = 'editor' in roles
+    is_promoter = 'promoter' in roles
+    is_usher = 'crew' in roles
+    twoflower_member = models.ProjectCrewMembership(
+        parent=project_expo2010,
+        user=user_twoflower,
+        is_editor=is_editor,
+        is_promoter=is_promoter,
+        is_usher=is_usher,
+        granted_by=user_vetinari,
+    )
+    twoflower_member.record_type = MEMBERSHIP_RECORD_TYPE.AMEND
+    db_session.add(twoflower_member)
+    db_session.commit()
+    return twoflower_member
```

Comment:
All these function names need a naming convention. If the `test_` prefix is unnecessary for pytest-bdd, we should use `given_`, `when_` and `then_` prefixes.